### PR TITLE
Allow Survey URL in connect-src for index CSP

### PR DESCRIPTION
**Pull request by kimlisa**
required in this PR https://github.com/gravitational/teleport/pull/38561. it's a questionnaire we show to new community users, and it sends the questionnaire info to this survey collecting URL `https://usage.teleport.dev`

### CODE REVIEW

**Comment by zmb3** [comment]
I don't recall doing this for teleport team, why is it necessary now?

Can the backend submit the survey instead?

File: lib/httplib/httplib_test.go [hunk 1]
```diff
@@ -374,6 +375,67 @@ func TestSetIndexContentSecurityPolicy(t *testing.T) {
 			}
 		})
 	}
+
+	indexCSPStringCache = newCSPCache()
```
Comment (ibeckermayer) [inline lib/httplib/httplib_test.go:379]: Why is this necessary?

File: lib/httplib/httpheaders.go [hunk 2]
```diff
@@ -179,6 +180,12 @@ func getIndexContentSecurityPolicy(withStripe, withWasm bool) cspMap {
 		cspMaps = append(cspMaps, wasmSecurityPolicy)
 	}
 
+	// Allows a URL for collecting surveys and feedback:
+	// https://github.com/gravitational/peopleware/blob/main/rfd/0001-adoption-metrics.md
+	if modules.GetModules().BuildType() == modules.BuildOSS {
+		cspMaps = append(cspMaps, cspMap{"connect-src": {"https://usage.teleport.dev"}})
```
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/httplib/httpheaders.go:186]: +1

File: lib/httplib/httplib_test.go [hunk 3]
```diff
@@ -374,6 +375,67 @@ func TestSetIndexContentSecurityPolicy(t *testing.T) {
...
+
+func TestSetIndexContentSecurityPolicy_OSSBuild(t *testing.T) {
+	modules.SetTestModules(t, &modules.TestModules{TestBuildType: modules.BuildOSS})
+
+	for _, tt := range []struct {
+		name            string
+		features        proto.Features
+		urlPath         string
+		expectedCspVals map[string]string
+		buildType       string
```
Comment (codingllama) [inline lib/httplib/httplib_test.go:390]: Unused?

```suggestion
```

File: lib/httplib/httplib_test.go [hunk 4]
```diff
@@ -374,6 +375,67 @@ func TestSetIndexContentSecurityPolicy(t *testing.T) {
...
+
+	indexCSPStringCache = newCSPCache()
+}
+
+func TestSetIndexContentSecurityPolicy_OSSBuild(t *testing.T) {
+	modules.SetTestModules(t, &modules.TestModules{TestBuildType: modules.BuildOSS})
+
+	for _, tt := range []struct {
+		name            string
+		features        proto.Features
```
Comment (codingllama) [inline lib/httplib/httplib_test.go:387]: Remove? We always seem to use the default.

```suggestion
```

File: lib/httplib/httplib_test.go [hunk 5]
```diff
@@ -374,6 +375,67 @@ func TestSetIndexContentSecurityPolicy(t *testing.T) {
...
+		urlPath         string
+		expectedCspVals map[string]string
+		buildType       string
+	}{
+		{
+			name:     "default (no stripe or wasm)",
```
Comment (codingllama) [inline lib/httplib/httplib_test.go:393]: I wonder if we should parameterize the tests with the buildType and join the tables into a single one. (It looks like you considered this too?)

File: lib/httplib/httplib_test.go [hunk 6]
```diff
@@ -374,6 +375,67 @@ func TestSetIndexContentSecurityPolicy(t *testing.T) {
...
+			features: proto.Features{},
+			urlPath:  "/web/index.js",
+			expectedCspVals: map[string]string{
```
Comment (codingllama) [inline lib/httplib/httplib_test.go:396]: Suggestion: create the "default" set of values and modify it for each test, so instead of a large map in each scenario we can instead how it changes compared to the default.

File: lib/httplib/httplib_test.go (hunk 1 above)
Comment (codingllama) [reply lib/httplib/httplib_test.go:379]: Same question - if it's a cleanup, then let's make sure it's done as such.

**Comment by kimlisa** [comment]
closing due to not planned
